--- a/source/externals/ptl/src/Threading.cc
+++ b/source/externals/ptl/src/Threading.cc
@@ -27,7 +27,7 @@
 #include "PTL/Utility.hh"
 
 #if defined(PTL_WINDOWS)
-#    include <Windows.h>
+#    include <windows.h>
 #endif
 
 #if defined(PTL_MACOS)
--- a/source/global/management/src/G4Threading.cc
+++ b/source/global/management/src/G4Threading.cc
@@ -35,7 +35,7 @@
 #include "globals.hh"
 
 #if defined(WIN32) || defined(__MINGW32__)
-#  include <Windows.h>
+#  include <windows.h>
 #else
 #  include <sys/syscall.h>
 #  include <sys/types.h>
--- a/source/global/management/src/G4FindDataDir.cc
+++ b/source/global/management/src/G4FindDataDir.cc
@@ -34,9 +34,6 @@
 #include <cstdlib>
 #include <cstring>
 
-#if defined(_MSC_VER)
-#define setenv(name, value, overwrite) _putenv_s(name, value)
-#endif
 
 using namespace G4fs;
 
@@ -84,7 +81,11 @@ static const char* G4FindDataDir(const char* name, const path& prefix, const pat
     else
       datadir /=  path(data_path) / dataset;
     if (is_directory(absolute(datadir)))
-      return setenv(name, absolute(datadir).string().c_str(), 0) == 0 ? std::getenv(name) : nullptr;
+//#if defined(_MSC_VER)
+      return _putenv_s(name, absolute(datadir).string().c_str()) == 0 ? std::getenv(name) : nullptr;
+//#else
+//      return setenv(name, absolute(datadir).string().c_str(), 0) == 0 ? std::getenv(name) : nullptr;
+//#endif
   }
--- a/source/global/management/src/G4Threading.cc
+++ b/source/global/management/src/G4Threading.cc
@@ -35,7 +35,7 @@
 
-#ifdef WIN32
+#ifdef _MSC_VER
 #define M_PI 3.141592653589793238463
 #endif
 
--- a/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
+++ b/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
@@ -11,7 +11,7 @@
 
 #include "nf_utilities.h"
 
-#ifdef WIN32
+#ifdef _MSC_VER
 #define M_PI 3.141592653589793238463
 #endif
 
--- a/source/processes/hadronic/models/lend/src/MCGIDI_energy.cc
+++ b/source/processes/hadronic/models/lend/src/MCGIDI_energy.cc
@@ -5,7 +5,7 @@
 #include <string.h>
 #include <cmath>
 
-#ifdef WIN32
+#ifdef _MSC_VER
 #define M_PI 3.141592653589793238463
 #endif
 