diff --git a/source/externals/ptl/src/Threading.cc b/source/externals/ptl/src/Threading.cc
index 324d6c12d7..a7553d8db2 100644
--- a/source/externals/ptl/src/Threading.cc
+++ b/source/externals/ptl/src/Threading.cc
@@ -27,7 +27,7 @@
 #include "PTL/Utility.hh"
 
 #if defined(PTL_WINDOWS)
-#    include <Windows.h>
+#    include <windows.h>
 #endif
 
 #if defined(PTL_MACOS)
diff --git a/source/global/management/src/G4FindDataDir.cc b/source/global/management/src/G4FindDataDir.cc
index 8c80180f71..7c5a6e0158 100644
--- a/source/global/management/src/G4FindDataDir.cc
+++ b/source/global/management/src/G4FindDataDir.cc
@@ -34,7 +34,7 @@
 #include <cstdlib>
 #include <cstring>
 
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #define setenv(name, value, overwrite) _putenv_s(name, value)
 #endif
 
diff --git a/source/global/management/src/G4Threading.cc b/source/global/management/src/G4Threading.cc
index 3c6bd77b14..fe09e672c7 100644
--- a/source/global/management/src/G4Threading.cc
+++ b/source/global/management/src/G4Threading.cc
@@ -35,7 +35,7 @@
 #include "globals.hh"
 
 #if defined(WIN32) || defined(__MINGW32__)
-#  include <Windows.h>
+#  include <windows.h>
 #else
 #  include <sys/syscall.h>
 #  include <sys/types.h>
diff --git a/source/processes/hadronic/models/lend/include/MCGIDI.h b/source/processes/hadronic/models/lend/include/MCGIDI.h
index e11303d2f4..3e4fa17887 100644
--- a/source/processes/hadronic/models/lend/include/MCGIDI.h
+++ b/source/processes/hadronic/models/lend/include/MCGIDI.h
@@ -9,7 +9,7 @@
 #define MCGIDI_VERSION_MINOR 0
 #define MCGIDI_VERSION_PATCHLEVEL 0
 
-#ifdef WIN32
+#ifdef _WIN32
 #define M_PI 3.141592653589793238463
 #endif
 
diff --git a/source/processes/hadronic/models/lend/include/nf_specialFunctions.h b/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
index 558091652f..6f70f1910d 100644
--- a/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
+++ b/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
@@ -10,7 +10,7 @@
 #include <float.h>
 #include <nf_utilities.h>
 
-#ifdef WIN32
+#ifdef _WIN32
 #define isfinite _finite
 #define M_PI 3.141592653589793238463
 /*#define INFINITY (DBL_MAX+DBL_MAX)*/
diff --git a/source/processes/hadronic/models/lend/src/MCGIDI_energy.cc b/source/processes/hadronic/models/lend/src/MCGIDI_energy.cc
index 65644c9727..d94d42eb78 100644
--- a/source/processes/hadronic/models/lend/src/MCGIDI_energy.cc
+++ b/source/processes/hadronic/models/lend/src/MCGIDI_energy.cc
@@ -5,7 +5,7 @@
 #include <string.h>
 #include <cmath>
 
-#ifdef WIN32
+#ifdef _WIN32
 #define M_PI 3.141592653589793238463
 #endif
 
diff --git a/source/processes/hadronic/models/lend/src/MCGIDI_outputChannel.cc b/source/processes/hadronic/models/lend/src/MCGIDI_outputChannel.cc
index 5894aec94d..3b7000dee3 100644
--- a/source/processes/hadronic/models/lend/src/MCGIDI_outputChannel.cc
+++ b/source/processes/hadronic/models/lend/src/MCGIDI_outputChannel.cc
@@ -2,6 +2,7 @@
 # <<BEGIN-copyright>>
 # <<END-copyright>>
 */
+#define _USE_MATH_DEFINES
 #include <string.h>
 #include <cmath>
 
